src/target/breakpoints.c

```diff
@@ -385,7 +438,8 @@ struct breakpoint *breakpoint_find(struct target *target, target_addr_t address)
 	struct breakpoint *breakpoint = target->breakpoints;
 
 	while (breakpoint) {
-		if (breakpoint->address == address)
+		if (breakpoint->address == address ||
+				(breakpoint->address == 0 && breakpoint->asid == address))
```

Comment:
@timsifive I have a few questions regarding this commit.

1. My understanding is that since this is a commit to the platform-independent part, we may have difficulties merging this upstream. Any chance we already have some MR to the upstream openocd? I searched on Gerrit, but could not find anything.

2. The reason I'm asking is that this condition `(breakpoint->address  0 && breakpoint->asid  address)` seems strange to me (since I think that `address` and `asid` should be independent entities). But it looks like this may be a defect of the existing OpenOCD interface, rather than this commit - if it is then it should be fixed upstream. So I was wondering if it makes sense to revise existing interface/data structures for setting breakpoints and if it makes sense to do it now, or if it is better to wait till these changes are upstreamed.

Reply:
There is no MR upstream. This change only makes sense when targets are unavailable, and that code is not present in upstream either.

Generally we make target-independent changes in this branch if they resolve issues that we see with RISC-V targets. (The change also needs to be reasonable so that some day it might be accepted upstream.)

I agree that comparing address to asid is odd, but that's also true upstream in breakpoint_remove_internal(). If this is causing problems on RISC-V, then feel to fix it here. If it's just a hypothetical bug that's not affecting anyone right now, I prefer for it to be fixed upstream.

Reply:
Got it, thanks for clarification!

> If it's just a hypothetical bug that's not affecting anyone right now, I prefer for it to be fixed upstream.

Currently, this is mostly a hypothetical bug... It's just I was scanning recent patches, and I was a little confused by these lines.